src/components/calcite-radio-button/calcite-radio-button.tsx

```diff
@@ -352,8 +361,10 @@ export class CalciteRadioButton implements LabelableComponent, CheckableFormComp
   };
 
   private onContainerFocus = (): void => {
-    this.focused = true;
-    this.calciteInternalRadioButtonFocus.emit();
+    if (!this.disabled) {
```

Comment:
Can you submit a separate PR with the `disabled` behavior-related changes (i.e., no styles)? You already did all of the heavy lifting, so I can take care of adding tests.

Reply:
If I do that, the accepted new focus snapshots will show up incorrect on the next Screener run.  Unless we're ok with accepting invalid visual states in order to merge this and then accept the new states in a follow-up PR?

Reply:
My suggestion was to hold on this one until the separate disabled PR lands (w/o styling changes). That wouldn't impact the visual states here, right?

Reply:
It will affect the disabled focus visual states.  Basically before this change, a disabled checkbox or radio-button would display a focus halo, and this corrects it so that when focusing a tile-select with `input-enabled` and `disabled` both on it won't show the focus halo in that specific case.

Reply:
Gotcha. I'm fine ignoring visual diffs from the other PR since this would get installed shortly after.

Reply:
Here are the fixes for this in a new PR: https://github.com/Esri/calcite-components/pull/3784

Reply:
Once that's merged I can update this branch that way we don't have to accept any incorrect snapshots.